fenix/app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayController.kt

```diff
@@ -541,10 +519,17 @@ class DefaultTabsTrayController(
     }
 
     override fun handleTabSelected(tab: TabSessionState, source: String?) {
-        TabsTray.openedExistingTab.record(TabsTray.OpenedExistingTabExtra(source ?: "unknown"))
-        tabsUseCases.selectTab(tab.id)
-        browsingModeManager.mode = BrowsingMode.fromBoolean(tab.content.private)
-        handleNavigateToBrowser()
+        val selected = tabsTrayStore.state.mode.selectedTabs
+        when {
+            selected.isEmpty() && tabsTrayStore.state.mode.isSelect().not() -> {
+                TabsTray.openedExistingTab.record(TabsTray.OpenedExistingTabExtra(source ?: "unknown"))
+                tabsUseCases.selectTab(tab.id)
+                browsingModeManager.mode = BrowsingMode.fromBoolean(tab.content.private)
+                handleNavigateToBrowser()
+            }
+            tab.id in selected.map { it.id } -> handleTabUnselected(tab)
```

Comment:
I'm wondering whether it's worth moving `handleTabUnselected` inline here as well. I see it's also used in the `Interactor` currently but I'm not sure if that's still necessary

Reply:
I'll take a look

Reply:
So it looks like `handleTabUnselected` is used exclusively for the [XML version of the tab reordering](https://searchfox.org/mozilla-mobile/source/firefox-android/fenix/app/src/main/java/org/mozilla/fenix/tabstray/browser/AbstractBrowserTabViewHolder.kt#251-257). We'll be able to clean this up after the XML gets removed though, so I'll file a follow-up.